include/silo/common/zstdfasta_reader.h

```diff
@@ -0,0 +1,37 @@
+#ifndef SILO_ZSTDFASTA_READER_H
+#define SILO_ZSTDFASTA_READER_H
+
+#include <filesystem>
+#include <iostream>
+
+#include <zstd.h>
+
+#include "silo/common/input_stream_wrapper.h"
+
+namespace silo {
+class ZstdFastaReader {
+  private:
+   silo::InputStreamWrapper in_file;
+   ZSTD_DDict* zstd_dictionary;
+   ZSTD_DCtx* zstd_context;
```

Comment:
Is it possible to use smart pointers here?

Reply:
Unfortunately not, as it is a C library, but good catch to add the relevant constructors, assignment operators and destructors to both classes

Reply:
Maybe a dumb question, but can't we wrap those raw pointers somehow? From the comment on `ZSTD_createDCtx` I suspect that those are supposed to be used as shared pointers in a sense?

Reply:
Yes, we could wrap them in a type that calls `ZSTD_freeDCtx` method in the destructor and then wrap then we also have the possibility to put that new type in a shared pointer, where multiple accesses to the dictionaries are possible. 
I guess this will create 4 files, but it removes the ugly constructors etc. from the FastaReader and Writer and can be reused in the streaming output for bulk downloading: #131 

Will do!

Reply:
For now the wrapper class itself is in a `unique_ptr` in the ZstdFastaReader / Writer, as we do not need to share it.
But we can as you correctly read from the comment also use it in `shared_ptr` when we need to